apps/andi/lib/andi_web/live/ingestion_live_view/edit_ingestion_live_view.ex

```diff
@@ -21,28 +22,78 @@ defmodule AndiWeb.IngestionLiveView.EditIngestionLiveView do
         <h2 class="component-title-text">Define Data Ingestion</h2>
       </div>
 
+      <div class="extract-steps-form-component">
+        <%= live_render(@socket, AndiWeb.IngestionLiveView.ExtractSteps.ExtractStepForm, id: :extract_step_form_editor, session: %{"ingestion" => @ingestion}) %>
+      </div>
+
       <div class="edit-page__btn-group">
+        <div class="btn-group__standard">
+          <button type="button" class="btn btn--large btn--cancel" phx-click="cancel-edit">Cancel</button>
+          <button id="save-button" name="save-button" class="btn btn--save btn--large" type="button" phx-click="save">Save Draft</button>
+        </div>
+
+        <hr>
+
         <button id="ingestion-delete-button" class="btn btn--delete" phx-click="prompt-ingestion-delete" type="button">
           <span class="delete-icon material-icons">delete_outline</span>
           DELETE
         </button>
+
       </div>
 
+      <%= live_component(@socket, AndiWeb.EditLiveView.UnsavedChangesModal, visibility: @unsaved_changes_modal_visibility) %>
       <%= live_component(@socket, AndiWeb.IngestionLiveView.DeleteIngestionModal, visibility: @delete_ingestion_modal_visibility) %>
+
+      <div id="edit-page-snackbar" phx-hook="showSnackbar">
+        <div style="display: none;"><%= @click_id %></div>
+          <%= if @save_success do %>
+            <div id="snackbar" class="success-message"><%= @success_message %></div>
+          <% end %>
+
+          <%= if @page_error do %>
+            <div id="snackbar" class="error-message">A page error occurred</div>
+          <% end %>
+      </div>
     </div>
     """
   end
 
   def mount(_params, %{"is_curator" => is_curator, "ingestion" => ingestion, "user_id" => user_id} = _session, socket) do
+    default_changeset = InputConverter.andi_ingestion_to_full_ui_changeset(ingestion)
+
     {:ok,
      assign(socket,
+       changeset: default_changeset,
+       click_id: nil,
+       delete_ingestion_modal_visibility: "hidden",
+       unsaved_changes_modal_visibility: "hidden",
        is_curator: is_curator,
-       user_id: user_id,
+       unsaved_changes: false,
+       page_error: false,
        ingestion: ingestion,
-       delete_ingestion_modal_visibility: "hidden"
+       save_success: false,
+       success_message: "",
+       user_id: user_id
      )}
   end
 
+  def handle_info(:form_update, socket) do
+    {:noreply, assign(socket, unsaved_changes: true)}
+  end
+
+  def handle_info({:update_save_message, status}, socket) do
+    message = save_message(status == "valid" && socket.assigns.changeset.valid?)
+
+    {:noreply, assign(socket, click_id: UUID.uuid4(), save_success: true, success_message: message)}
+  end
+
+  # This handle_info takes care of all exceptions in a generic way.
+  # Expected errors should be handled in specific handlers.
+  # Flags should be reset here.
+  def handle_info({:EXIT, _pid, {_error, _stacktrace}}, socket) do
```

Comment:
What does :EXIT do? it seems like it kills the process? Is that desired?

Are those flags for showing toasts or on-screen errors to the user?

Reply:
@jfolk2015 - this is a great question. This handle_info event is boilerplate code that exists in several of the views. From what we can tell, this seems to show an error message on screen to the viewer if it's encountered. (Specifically, there's a small modal that pops up at the bottom of the page that says," A page error occurred.") @LtChae - do you have any additional thoughts here?